src/aro/Parser.zig

```diff
@@ -7683,6 +7684,18 @@ fn callExpr(p: *Parser, lhs: Result) Error!Result {
             continue;
         }
         const p_ty = params[arg_count].ty;
+        if (p_ty.statically_sized_array) {
+            if (arg.ty.data == .array and arg.ty.data.array.*.len < p_ty.data.array.*.len) {
+                const extra = Diagnostics.Message.Extra{ .arguments = .{
+                    .expected = @intCast(arg.ty.data.array.*.len),
+                    .actual = @intCast(p_ty.data.array.*.len),
+                } };
+                try p.errExtra(.array_argument_too_small, param_tok, extra);
```

Comment:
Clang also adds a `callee declares array parameter as static here` note. You should be able to do the same with `params[arg_count].name_tok`.